tests/functional/test_manager.py

```diff
@@ -125,3 +139,12 @@ def test_enable_disable_callbacks_only_ever_registered_once(self):
             body.disable_callback_call_count, 1,
             'The disable_callback() should have only ever been registered '
             'once')
+
+    def test_use_custom_executor_implementation(self):
+        transfer_manager = TransferManager(
+            self.client, executor_cls=BaseExecutor)
+        # The BaseExecutor class has no methods implemented so it should
+        # immediately raise a NotImplementedError once something is
+        # submitted to it
+        with self.assertRaises(NotImplementedError):
+            transfer_manager.delete('bucket', 'key')
```

Comment:
I don't think this is necessarily a strict contract we have.  I.e if we decide to move implementation into the base class, that's perfectly reasonable, and yet this test will fail.

I think it would better to just pass in a mock for `executor_cls` here.